### DISCUSSION TITLE
Locust to test network SD-WAN performance

### DISCUSSION THREAD

**Discussion by scottmillers** [locustio/locust#2808] [General]
Problem:
- We want to simulate the network load for our web application using Locust.  
- We don't have access to the web applications source code and can't deploy it. 

Solution:  
- A locust example with a corresponding web application we can use to simulate load on the network 

Does this exist?

**Comment by cyberw** [comment]
Hi! I dont quite understand the question. Do you want to test the web application, or just the network?

And you say you cant deploy the web application, but you want a "corresponding web application"?

**Comment by scottmillers** [comment]
Hey Lars, the goal is to test the network.  The network uses software that optimizes traffic through caching and other means.  We want to put a load on this network that simulates our application traffic.  We don't have access to the application.  We are looking for a solution with a sample web application, that we can deploy to AWS, and a set of test use cases written in Locust.  We run the tests in parallel to simulate our current traffic load.

**Comment by cyberw** [comment]
I see! No, I dont know of any sample web application, but maybe you can just set up a basic webserver?

Depending on what kind of traffic (big/small payloads, simulated fast/slow response times etc) you want to be sent over your network you'll need to customize it anyway.

If you're only interested in putting load on the network there are other tools more geared towards that, like iperf https://iperf.fr/
